core/src/main/java/jenkins/model/RunIdMigrator.java

```diff
@@ -118,14 +118,15 @@ private boolean load(File dir) {
     private void save(File dir) {
         File f = new File(dir, MAP_FILE);
         try {
-            AtomicFileWriter w = new AtomicFileWriter(f);
-            try {
-                for (Map.Entry<String,Integer> entry : idToNumber.entrySet()) {
-                    w.write(entry.getKey() + ' ' + entry.getValue() + '\n');
+            try (AtomicFileWriter w = new AtomicFileWriter(f)) {
+                try {
```

Comment:
I doubt new `try` is needed here, add to the one above?

Reply:
Thanks, I merged them as suggested.